components/tikv_util/src/yatp_pool/mod.rs

```diff
@@ -192,13 +227,34 @@ impl<T: PoolTicker> Runner for YatpPoolRunner<T> {
 
     fn handle(&mut self, local: &mut Local<Self::TaskCell>, mut task_cell: Self::TaskCell) -> bool {
         let extras = task_cell.mut_extras();
-        if let Some(schedule_time) = extras.schedule_time() {
-            let idx = priority_from_task_meta(extras.metadata()) as usize;
-            self.schedule_wait_durations[idx].observe(schedule_time.elapsed().as_secs_f64());
+        let priority = priority_from_task_meta(extras.metadata());
+        let start_time =
+            if self.schedule_wait_durations.enabled() || self.schedule_exec_durations.enabled() {
+                Some(std::time::Instant::now())
+            } else {
+                None
+            };
+        if let Some(dur) = start_time.and_then(|t1| {
+            extras
+                .schedule_time()
+                .map(|t2| t1.saturating_duration_since(t2))
+        }) {
```

Comment:
We can match 2 `Option`s together by `zip`.
```suggestion
        if let Some(dur) = start_time
            .zip(extras.schedule_time())
            .map(|(t1, t2)| t1.saturating_duration_since(t2))
        {
```